pyo3-macros-backend/src/intopyobject.rs

```diff
@@ -389,8 +431,21 @@ impl<'a> Container<'a> {
                     .map(|item| item.value())
                     .unwrap_or_else(|| f.ident.unraw().to_string());
                 let value = Ident::new(&format!("arg{i}"), f.field.ty.span());
-                quote! {
-                    #pyo3_path::types::PyDictMethods::set_item(&dict, #key, #value)?;
+                let expr_path = if REF {
+                    f.into_py_with_ref.as_ref().map(|i|&i.value)
+                } else {
+                    f.into_py_with.as_ref().map(|i|&i.value)
+                };
+
+                if let Some(expr_path) = expr_path {
+                    quote! {
+                        let into_py_with: fn(_, #pyo3_path::Python<'_>) -> #pyo3_path::PyResult<#pyo3_path::Bound<'_, #pyo3_path::PyAny>> = #expr_path;
```

Comment:
👍 nice use of the same trick from #4840! Maybe add a UI test to help keep nice diagnostics?

Reply:
I think I needed it for a different reason, but if it helps with the diagnostics even better 😁. Will add a ui test.

Reply:
Not quite as nice as in #4840, since it still talks about "fn item", but still quite readable i would say.